google/resource_storage_notification_test.go

```diff
@@ -0,0 +1,250 @@
+						bucketName, "google_storage_notification.notification", &notification),
+					resource.TestCheckResourceAttr(
+						"google_storage_notification.notification", "bucket", bucketName),
+					resource.TestCheckResourceAttr(
+						"google_storage_notification.notification", "topic", topic),
+					resource.TestCheckResourceAttr(
+						"google_storage_notification.notification", "payload_format", payload),
+					resource.TestCheckResourceAttr(
+						"google_storage_notification.notification_with_prefix", "object_name_prefix", "foobar"),
+				),
+			},
+			resource.TestStep{
+				ResourceName:      "google_storage_notification.notification",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+			resource.TestStep{
+				ResourceName:      "google_storage_notification.notification_with_prefix",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func TestAccGoogleStorageNotification_withEventsAndAttributes(t *testing.T) {
+	t.Parallel()
+
+	skipIfEnvNotSet(t, "GOOGLE_PROJECT")
+
+	var notification storage.Notification
+	bucketName := testBucketName()
+	topicName := fmt.Sprintf("tf-pstopic-test-%d", acctest.RandInt())
+	topic := fmt.Sprintf("projects/%s/topics/%s", os.Getenv("GOOGLE_PROJECT"), topicName)
+	eventType1 := "OBJECT_FINALIZE"
+	eventType2 := "OBJECT_ARCHIVE"
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccGoogleStorageNotificationDestroy,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testGoogleStorageNotificationOptionalEventsAttributes(bucketName, topicName, topic, eventType1, eventType2),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckStorageNotificationExists(
+						bucketName, "google_storage_notification.notification", &notification),
+					resource.TestCheckResourceAttr(
+						"google_storage_notification.notification", "bucket", bucketName),
+					resource.TestCheckResourceAttr(
+						"google_storage_notification.notification", "topic", topic),
+					resource.TestCheckResourceAttr(
+						"google_storage_notification.notification", "payload_format", payload),
+					testAccCheckStorageNotificationCheckEventType(
+						&notification, []string{eventType1, eventType2}),
+					testAccCheckStorageNotificationCheckAttributes(
+						&notification, "new-attribute", "new-attribute-value"),
+				),
+			},
+			resource.TestStep{
+				ResourceName:      "google_storage_notification.notification",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func testAccGoogleStorageNotificationDestroy(s *terraform.State) error {
+	config := testAccProvider.Meta().(*Config)
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "google_storage_notification" {
+			continue
+		}
+
+		bucket := rs.Primary.Attributes["bucket"]
+
+		_, notificationID := resourceStorageNotificationParseID(rs.Primary.ID)
```

Comment:
Since the bucket is in the id also, you could just do `bucket, notificationID` here